resources/certificate.rb

```diff
@@ -107,19 +123,21 @@
 end
 
 action_class do
+  require 'openssl'
+
   def fetch_certdata
     return IO.read(new_resource.cert_file) unless new_resource.cert_file.nil?
 
     certendpoint = new_resource.ssl_endpoint
     unless certendpoint.nil?
-      cmd = Mixlib::ShellOut.new("echo QUIT | openssl s_client -showcerts -connect #{certendpoint} 2> /dev/null | openssl x509")
-      cmd.run_command
-      Chef::Log.debug(cmd.format_for_exception)
-
-      Chef::Application.fatal!("Error returned when attempting to retrieve certificate from remote endpoint #{certendpoint}: #{cmd.exitstatus}", cmd.exitstatus) unless cmd.exitstatus == 0
-
-      certout = cmd.stdout
-      return certout unless certout.empty?
+      ctx = OpenSSL::SSL::SSLContext.new
```

Comment:
This is a breaking change and one I think needs some discussion before doing so, `openssl` is packaged in every major distribution (including fedora) as there are some differences between the ruby implementation and the package provided by the OS I think we need to be more careful and do this as a separate prs so if it causes issues with users we can roll back just the openssl change keeping the rest of the changes in place. For testing you can use an intermediate instruction in `kitchen-dokken`.

Reply:
I will remove this change